test/libsolidity/util/BytesUtils.cpp

```diff
@@ -164,35 +201,105 @@ string BytesUtils::formatString(bytes const& _bytes) const
 	return os.str();
 }
 
-bytes BytesUtils::alignLeft(bytes _bytes) const
+string BytesUtils::formatRawBytes(bytes const& _bytes)
 {
-	return std::move(_bytes) + bytes(32 - _bytes.size(), 0);
+	if (!_bytes.empty())
+	{
+		stringstream os;
+		auto it = _bytes.begin();
+		size_t size = 32;
+		for (size_t i = 0; i < _bytes.size() / size; i++)
+		{
+			long offset = static_cast<long>(size);
+			auto offsetIter = it + offset;
+			bytes byteRange{it, offsetIter};
+
+			os << "  " << byteRange;
+
+			it += offset;
+			if (it != _bytes.end())
+				os << endl;
+		}
+
+		return os.str();
+	}
+	else
+		return "[]";
 }
 
-bytes BytesUtils::alignRight(bytes _bytes) const
+string BytesUtils::formatBytes(
+	bytes const& _bytes,
+	ABIType const& _abiType
+)
 {
-	return bytes(32 - _bytes.size(), 0) + std::move(_bytes);
+	stringstream os;
+
+	switch (_abiType.type)
+	{
+	case ABIType::UnsignedDec:
+		// Check if the detected type was wrong and if this could
+		// be signed. If an unsigned was detected in the expectations,
+		// but the actual result returned a signed, it would be formatted
+		// incorrectly.
+		if (*_bytes.begin() & 0x80)
```

Comment:
@chriseth Fixed https://github.com/ethereum/solidity/pull/7048#discussion_r301513425 here. `formatUnsigned` does not do this "auto-correction" anymore and I moved it here to make it even more clear.